gcp/apis/database/v1alpha2/cloudsql_instance_types.go

```diff
@@ -179,7 +197,8 @@ type CloudsqlInstanceList struct {
 // ConnectionSecret returns a connection secret for this instance
 func (i *CloudsqlInstance) ConnectionSecret() *corev1.Secret {
 	s := resource.ConnectionSecretFor(i, CloudsqlInstanceGroupVersionKind)
-	s.Data[runtimev1alpha1.ResourceCredentialsSecretEndpointKey] = []byte(i.Status.Endpoint)
+	s.Data[PublicIPKey] = []byte(i.Status.PublicIP)
+	s.Data[PrivateIPKey] = []byte(i.Status.PrivateIP)
```

Comment:
I suggest we also save the public IP under the `endpoint` key for compatibility with the current behaviour. The `endpoint` concept is probably too simplistic going forward, but it's currently set by every controller that writes a secret.